frontend_tests/puppeteer_lib/common.ts

```diff
@@ -494,6 +494,9 @@ class CommonUtils {
         const page = await this.get_page();
         try {
             await test_function(page);
+            // Clear all the server-side cache, because we are resetting the
+            // test environment after each run.
+            await page.goto(this.realm_url + "flush_caches");
```

Comment:
It might make sense to do this HTTP request from `test-js-with-puppeteer` via the `requests` module, so that one can control when the flushes happen in relation to the other test teardown steps?

Though we might want to have a step here that closes the browser window or the like, to avoid presence requests being sent to the zulip server while we're resetting its database.

Reply:
We already do that,
```python
# frontend_tests/puppeteer_lib/common.ts
509        } finally {
510           await browser.close();
511        }
```